internal/terminal/table_test.go

```diff
@@ -0,0 +1,264 @@
+			expectedMessage: "\n" + fmt.Sprintf("%s  %s  %s  %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("header"),
+					color.New(color.Bold).SprintFunc()("only"),
+					color.New(color.Bold).SprintFunc()("no"),
+					color.New(color.Bold).SprintFunc()("data"),
+				}...,
+			) + `
+------  ----  --  ----
+`,
+		},
+		{
+			description: "Should return a string representing the table",
+			header:      []string{"one", "two", "three"},
+			data: []map[string]interface{}{
+				{"one": 1, "two": 2.222, "three": "three"},
+				{"one": "1", "two": 2, "three": 3},
+				{"one": "1.0", "two": 2, "three": 3},
+			},
+
+			expectedMessage: "\n" + fmt.Sprintf("%s  %s    %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("one"),
+					color.New(color.Bold).SprintFunc()("two"),
+					color.New(color.Bold).SprintFunc()("three")}...,
+			) + `
+---  -----  -----
+1    2.222  three
+1    2      3    
+1.0  2      3    `,
+		},
+		{
+			description: "Should return rows with values specified in the header, not insert new columns, and not print empty rows",
+			header:      []string{"three", "one", "two"},
+			data: []map[string]interface{}{
+				{},
+				{"one": 1, "two": 2, "three": -3.123},
+				{},
+				{"one": 1, "two": 2, "test": "no"},
+				{},
+				{"one": 1, "should not show up": 2},
+				{},
+				{"two": []int{2, 3}},
+			},
+
+			expectedMessage: "\n" + fmt.Sprintf("%s   %s  %s  ",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("three"),
+					color.New(color.Bold).SprintFunc()("one"),
+					color.New(color.Bold).SprintFunc()("two")}...,
+			) + `
+------  ---  -----
+-3.123  1    2    
+        1    2    
+        1         
+             [2 3]`,
+		},
+		{
+			description: "Should return a table where the longest length is determined by the longest value in the column as a whole",
+			header:      []string{"short", "longer header that will determine length", "short again"},
+			data: []map[string]interface{}{
+				{"short": 1, "longer header that will determine length": "tiny", "short again": 123},
+				{"short": "the word short is actually longer than the word long", "longer header that will determine length": "tiny again", "short again": 90},
+			},
+			expectedMessage: "\n" + fmt.Sprintf("%s                                                 %s  %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("short"),
+					color.New(color.Bold).SprintFunc()("longer header that will determine length"),
+					color.New(color.Bold).SprintFunc()("short again")}...,
+			) + `
+----------------------------------------------------  ----------------------------------------  -----------
+1                                                     tiny                                      123        
+the word short is actually longer than the word long  tiny again                                90         `,
+		},
+		{
+			description: "Should return correctly formatted values in the table",
```

Comment:
this test case should be duplicated somehow, but with `color.DisableColor = true` set at the start of one of the tests. then we can assert the string is printed without bold formatting as well